src/asl.ts

```diff
@@ -428,21 +428,52 @@ export class ASL {
           return hoist && self.doHoist(node)
             ? hoist(updatedChild)
             : updatedChild;
+        } else if (isBinaryExpr(node)) {
+          const updated = visitEachChild(node, (expr) =>
+            normalizeAST(expr, hoist)
+          );
+          return hoist && self.doHoist(node) ? hoist(updated) : updated;
+        } else if (
+          isArrayLiteralExpr(node) &&
+          !node.items.some((i) => isFunctionExpr(i))
+        ) {
+          // if the array contains functions, just leave it, it is probably for Parallel.
+          // This is short sighted, but should work now
```

Comment:
do we need an issue to track this?

Reply:
I'm changing this. In fact, I am rewriting all of it. Lol. I hope.